modules/ROOT/pages/rest-sdk/valueresolver.adoc

```diff
@@ -0,0 +1,250 @@
+can be obtained by making a request to another endpoint:
+https://api.slack.com/methods/conversations.list
+
+image:media/image2.png[image,width=624,height=328]
+
+A response from this endpoint looks like this:
+
+[source,yaml]
+----
+{
+    "ok": true,
+    "channels": [
+        {
+            "id": "C012AB3CD",
+            "name": "general", [..]
+        },
+        {
+            "id": "C061EG9T2",
+            "name": "random", [..]
+        },
+        [..]
+    ],
+    "response_metadata": {
+        "next_cursor": "dGVhbTpDMDYxRkE1UEI="
+    }
+}
+----
+
+Given this, you can create a simple value resolver that enables you to
+provide the possible values for the `channels` parameter.
+
+[source,yaml]
+----
+valueResolvers:
+ channels:
+   path: /conversations.list
+   method: GET
+   items:
+      extraction:
+         expression: “#[payload.channels]”
+      displayName:
+         expression: “#[item.name]”
+      value:
+         expression: “#[item.id]”
+----
+
+According to the example, the value resolver must make a `GET` request
+to the `/conversations.list` endpoint using the path and method
+properties respectively.
+
+Next, the resolver extracts the array of values from the response
+with the `extraction` expression, and for each one of these items, `name` property is returned as *displayName*, and the `id` property is returned as *value*.
+
+The user sees *displayName* in the dropdown for the
+possible channel values, and *value* is sent to the server.
+
+You can then add this value resolver to the channel
+parameter in the operation as shown:
+
+[source,yaml]
+----
+endpoints:
+ /chat.postMessage:
+   operations:
+     post:
+       expects:
+         queryParameter:
+           channel:
+             valueResolver:
+               id: channels
+----
+
+== Example: Value Resolver with Parameters
+
+This example contains the previous parameter resolver and adds
+some intelligence. This value resolver uses the operation’s
+parameters in order to query for the appropriate type of channels.
+
+[source,yaml]
+----
+valueResolvers:
+ channels:
+   path: /conversations.list
+   method: GET
+   parameters:
+      channelType:
+        type: string
+   binding:
+     queryParams:
+       channelType:
+         value:
+           expression: “#[if(parameters.channelType != null) parameters.channelType else ‘all’]”
+   items:
+      extraction:
+         expression: “#[payload.channels]”
+      displayName:
+         expression: “#[item.name]”
+      value:
+         expression: “[item.id]”
+----
+
+This value resolver resembles the previous one with the exception of
+the addition of the `binding` node. +
+You can use this node to bind parameter values from the operation’s where
+this value resolver is used to the request that is made to
+request the possible values.
+// I don't really understand what this sentence is trying to convey ^ Is operations supposed to be possessive?
```

Comment:
This is a way of "forwarding" parameters set in the operation to the value resolver, so it can use those values to make a request and get the possible values the parameter is allowed to have.

Reply:
```suggestion
```